Sporadic test failure: TestBulkDocsUnusedSequencesMultiRevDoc

## TestBulkDocsUnusedSequencesMultiRevDoc

### Repro

http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration/27/console

### Logs

```
=== RUN   TestBulkDocsUnusedSequencesMultiRevDoc
2017-10-19T12:42:52.636-07:00 Using xattrs: strings.ToLower(useXattrs) == strings.ToLower(TestEnvSyncGatewayTrue).  |true| == |true|
2017-10-19T12:42:52.637-07:00 WARNING: Error processing DCP stream - will attempt to restart/reconnect: pkt.Receive, err: read tcp 172.23.105.143:36454->172.23.109.96:11210: use of closed network connection. -- base.(*DCPReceiver).OnError() at dcp_feed.go:120
2017-10-19T12:42:54.224-07:00 GoCBCustomSGTranscoder Opening Couchbase database test_data_bucket on <http://integration-cbs.sc.couchbase.com:8091> as user "test_data_bucket"
2017-10-19T12:42:54.236-07:00 Using xattrs: strings.ToLower(useXattrs) == strings.ToLower(TestEnvSyncGatewayTrue).  |true| == |true|
2017-10-19T12:42:54.236-07:00 Using xattrs: strings.ToLower(useXattrs) == strings.ToLower(TestEnvSyncGatewayTrue).  |true| == |true|
2017-10-19T12:42:54.236-07:00 Opening db /db as bucket "test_data_bucket", pool "default", server <http://integration-cbs.sc.couchbase.com:8091>
2017-10-19T12:42:54.236-07:00 GoCBCustomSGTranscoder Opening Couchbase database test_data_bucket on <http://integration-cbs.sc.couchbase.com:8091> as user "test_data_bucket"
2017-10-19T12:42:54.575-07:00 Cache: Initializing changes cache with options {ChannelCacheOptions:{ChannelCacheMinLength:0 ChannelCacheMaxLength:0 ChannelCacheAge:0s} CachePendingSeqMaxWait:5s CachePendingSeqMaxNum:10000 CacheSkippedSeqMaxWait:1h0m0s}
2017-10-19T12:42:54.614-07:00 Using metadata purge interval of 3.00 days for tombstone compaction.
_time=2017-10-19T12:42:54.628-07:00 _level=INFO _msg=Using plain authentication for user test_data_bucket 
2017-10-19T12:42:54.703-07:00 Changes+: Notifying that "test_data_bucket" changed (keys="{_sync:user:}") count=2
2017-10-19T12:42:54.966-07:00 HTTP:  #247: POST http://localhost/db/_bulk_docs
2017-10-19T12:42:54.966-07:00 Changes+: Notifying that "test_data_bucket" changed (keys="{_sync:user:}") count=3
2017-10-19T12:42:54.970-07:00 WARNING: Sync fn exception: Rejecting invalid doc; doc = map[n:%!s(float64=2) type:invalid _rev:1-a69b6e0644577af07d30a6b5c3d00fd7 _id:bulk2] -- db.(*Database).getChannelsAndAccess() at crud.go:1199
2017-10-19T12:42:54.970-07:00 	BulkDocs: Doc "bulk2" --> 500 Exception in JS sync function (500 Exception in JS sync function)
2017-10-19T12:42:54.971-07:00 Cache: Received #1 after   1ms ("bulk1" / "1-50133ddd8e49efad34ad9ecae4cb9907")
2017-10-19T12:42:54.971-07:00 Cache: Initialized cache for channel "*" with options: &{ChannelCacheMinLength:50 ChannelCacheMaxLength:500 ChannelCacheAge:1m0s}
2017-10-19T12:42:54.971-07:00 Cache:     #1 ==> channel "*"
2017-10-19T12:42:54.971-07:00 Changes+: Notifying that "test_data_bucket" changed (keys="{*}") count=4
2017-10-19T12:42:54.972-07:00 Cache: Received #2 after   0ms ("bulk3" / "1-bfa09fe957e5e28b750cff0568b1cb05")
2017-10-19T12:42:54.972-07:00 Cache:     #2 ==> channel "*"
2017-10-19T12:42:54.972-07:00 Changes+: Notifying that "test_data_bucket" changed (keys="{*}") count=5
2017-10-19T12:42:54.973-07:00 Using xattrs: strings.ToLower(useXattrs) == strings.ToLower(TestEnvSyncGatewayTrue).  |true| == |true|
2017-10-19T12:42:54.973-07:00 Opening db /db as bucket "test_data_bucket", pool "default", server <http://integration-cbs.sc.couchbase.com:8091>
2017-10-19T12:42:54.973-07:00 GoCBCustomSGTranscoder Opening Couchbase database test_data_bucket on <http://integration-cbs.sc.couchbase.com:8091> as user "test_data_bucket"
2017-10-19T12:42:55.312-07:00 Cache: Initializing changes cache with options {ChannelCacheOptions:{ChannelCacheMinLength:0 ChannelCacheMaxLength:0 ChannelCacheAge:0s} CachePendingSeqMaxWait:5s CachePendingSeqMaxNum:10000 CacheSkippedSeqMaxWait:1h0m0s}
2017-10-19T12:42:55.333-07:00 **NOTE:** "db"'s sync function has changed. The new function may assign different channels to documents, or permissions to users. You may want to re-sync the database to update these.
2017-10-19T12:42:55.333-07:00 Using default sync function 'channel(doc.channels)' for database "db"
2017-10-19T12:42:55.340-07:00 HTTP:  #248: POST http://localhost/db/_bulk_docs
2017-10-19T12:42:55.344-07:00 Cache: Received #7 after   0ms ("bulk1" / "2-d2111723d973efbcc415aab0ade25a45")
2017-10-19T12:42:55.344-07:00 Cache:   Deferring #7 (1 now waiting for #3...#6)

	api_test.go:966
	assert.Equals(t, doc1Rev2.Sequence, uint64(7))	

	expected: 0x7
	     got: 0x0
2017-10-19T12:42:55.347-07:00 HTTP:  #249: POST http://localhost/db/_bulk_docs
2017-10-19T12:42:55.348-07:00 	BulkDocs: Doc "bulk1" --> 409 Document exists (409 Document exists)

	api_test.go:982
	assert.Equals(t, doc1Rev3.Sequence, uint64(8))	

	expected: 0x8
	     got: 0x7

	api_test.go:986
	assert.Equals(t, len(rs), 3)	

	expected: 3
	     got: 2
2017-10-19T12:42:55.350-07:00 Changes+: changeListener.Stop() called
2017-10-19T12:42:55.350-07:00 Changes+: changeListener.Stop() called
--- FAIL: TestBulkDocsUnusedSequencesMultiRevDoc (2.71s)
	assert.go:31: 
	assert.go:31: 
	assert.go:31: 
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 4040 [running]:
panic(0xbb28a0, 0xc4200120f0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc42109ea80)
	/usr/local/go/src/testing/testing.go:579 +0x25d
panic(0xbb28a0, 0xc4200120f0)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/couchbase/sync_gateway/rest.TestBulkDocsUnusedSequencesMultiRevDoc(0xc42109ea80)
	/var/jenkins/workspace/sync-gateway-integration/godeps/src/github.com/couchbase/sync_gateway/rest/api_test.go:989 +0x144f
testing.tRunner(0xc42109ea80, 0xd5af58)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec
exit status 2
FAIL	github.com/couchbase/sync_gateway/rest	158.907s
```

--- a/rest/api_test.go
+++ b/rest/api_test.go
@@ -786,14 +786,17 @@ func TestBulkDocsUnusedSequences(t *testing.T) {
 	response := rt.SendRequest("POST", "/db/_bulk_docs", input)
 	assertStatus(t, response, 201)
 
-	doc1Rev, _ := rt.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev, err := rt.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev.Sequence, uint64(1))
 
-	doc3Rev, _ := rt.GetDatabase().GetDocSyncData("bulk3")
+	doc3Rev, err := rt.GetDatabase().GetDocSyncData("bulk3")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc3Rev.Sequence, uint64(2))
 
 	//Get current sequence number, this will be 3, as SG allocates enough sequences to process all bulk docs
-	lastSequence, _ := rt.GetDatabase().LastSequence()
+	lastSequence, err := rt.GetDatabase().LastSequence()
+	assertNoError(t, err, "LastSequence error")
 	assert.Equals(t, lastSequence, uint64(3))
 
 	//send another _bulk_docs and validate the sequences used
@@ -802,17 +805,21 @@ func TestBulkDocsUnusedSequences(t *testing.T) {
 	assertStatus(t, response, 201)
 
 	//Sequence 3 get used here
-	doc21Rev, _ := rt.GetDatabase().GetDocSyncData("bulk21")
+	doc21Rev, err := rt.GetDatabase().GetDocSyncData("bulk21")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc21Rev.Sequence, uint64(3))
 
-	doc22Rev, _ := rt.GetDatabase().GetDocSyncData("bulk22")
+	doc22Rev, err := rt.GetDatabase().GetDocSyncData("bulk22")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc22Rev.Sequence, uint64(4))
 
-	doc23Rev, _ := rt.GetDatabase().GetDocSyncData("bulk23")
+	doc23Rev, err := rt.GetDatabase().GetDocSyncData("bulk23")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc23Rev.Sequence, uint64(5))
 
 	//Get current sequence number
-	lastSequence, _ = rt.GetDatabase().LastSequence()
+	lastSequence, err = rt.GetDatabase().LastSequence()
+	assertNoError(t, err, "LastSequence error")
 	assert.Equals(t, lastSequence, uint64(5))
 }
 
@@ -826,14 +833,17 @@ func TestBulkDocsUnusedSequencesMultipleSG(t *testing.T) {
 	response := rt1.SendRequest("POST", "/db/_bulk_docs", input)
 	assertStatus(t, response, 201)
 
-	doc1Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev, err := rt1.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev.Sequence, uint64(1))
 
-	doc3Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk3")
+	doc3Rev, err := rt1.GetDatabase().GetDocSyncData("bulk3")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc3Rev.Sequence, uint64(2))
 
 	//Get current sequence number, this will be 3, as SG allocates enough sequences to process all bulk docs
-	lastSequence, _ := rt1.GetDatabase().LastSequence()
+	lastSequence, err := rt1.GetDatabase().LastSequence()
+	assertNoError(t, err, "LastSequence error")
 	assert.Equals(t, lastSequence, uint64(3))
 
 	rt2 := RestTester{RestTesterBucket: rt1.RestTesterBucket, SyncFn: `function(doc) {if(doc.type == "invalid") {throw("Rejecting invalid doc")}}`}
@@ -849,7 +859,7 @@ func TestBulkDocsUnusedSequencesMultipleSG(t *testing.T) {
 	spec := base.GetTestBucketSpec(base.DataBucket)
 	username, password, _ := spec.Auth.GetCredentials()
 
-	_, err := rt2.RestTesterServerContext.AddDatabaseFromConfig(&DbConfig{
+	_, err = rt2.RestTesterServerContext.AddDatabaseFromConfig(&DbConfig{
 		BucketConfig: BucketConfig{
 			Server:   &server,
 			Bucket:   &bucketName,
@@ -867,17 +877,21 @@ func TestBulkDocsUnusedSequencesMultipleSG(t *testing.T) {
 	assertStatus(t, response, 201)
 
 	//Sequence 3 does not get used here as its using a different sequence allocator
-	doc21Rev, _ := rt2.GetDatabase().GetDocSyncData("bulk21")
+	doc21Rev, err := rt2.GetDatabase().GetDocSyncData("bulk21")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc21Rev.Sequence, uint64(4))
 
-	doc22Rev, _ := rt2.GetDatabase().GetDocSyncData("bulk22")
+	doc22Rev, err := rt2.GetDatabase().GetDocSyncData("bulk22")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc22Rev.Sequence, uint64(5))
 
-	doc23Rev, _ := rt2.GetDatabase().GetDocSyncData("bulk23")
+	doc23Rev, err := rt2.GetDatabase().GetDocSyncData("bulk23")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc23Rev.Sequence, uint64(6))
 
 	//Get current sequence number
-	lastSequence, _ = rt2.GetDatabase().LastSequence()
+	lastSequence, err = rt2.GetDatabase().LastSequence()
+	assertNoError(t, err, "LastSequence error")
 	assert.Equals(t, lastSequence, uint64(6))
 
 	//Now send a bulk_doc to rt1 and see if it uses sequence 3
@@ -886,13 +900,16 @@ func TestBulkDocsUnusedSequencesMultipleSG(t *testing.T) {
 	assertStatus(t, response, 201)
 
 	//Sequence 3 get used here as its using first sequence allocator
-	doc31Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk31")
+	doc31Rev, err := rt1.GetDatabase().GetDocSyncData("bulk31")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc31Rev.Sequence, uint64(3))
 
-	doc32Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk32")
+	doc32Rev, err := rt1.GetDatabase().GetDocSyncData("bulk32")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc32Rev.Sequence, uint64(7))
 
-	doc33Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk33")
+	doc33Rev, err := rt1.GetDatabase().GetDocSyncData("bulk33")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc33Rev.Sequence, uint64(8))
 
 }
@@ -908,17 +925,20 @@ func TestBulkDocsUnusedSequencesMultiRevDoc(t *testing.T) {
 	response := rt1.SendRequest("POST", "/db/_bulk_docs", input)
 	assertStatus(t, response, 201)
 
-	doc1Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev, err := rt1.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev.Sequence, uint64(1))
 
 	//Get the revID for doc "bulk1"
 	doc1RevID := doc1Rev.CurrentRev
 
-	doc3Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk3")
+	doc3Rev, err := rt1.GetDatabase().GetDocSyncData("bulk3")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc3Rev.Sequence, uint64(2))
 
 	//Get current sequence number, this will be 3, as SG allocates enough sequences to process all bulk docs
-	lastSequence, _ := rt1.GetDatabase().LastSequence()
+	lastSequence, err := rt1.GetDatabase().LastSequence()
+	assertNoError(t, err, "LastSequence error")
 	assert.Equals(t, lastSequence, uint64(3))
 
 	rt2 := RestTester{RestTesterBucket: rt1.RestTesterBucket, SyncFn: `function(doc) {if(doc.type == "invalid") {throw("Rejecting invalid doc")}}`}
@@ -935,7 +955,7 @@ func TestBulkDocsUnusedSequencesMultiRevDoc(t *testing.T) {
 	username, password, _ := spec.Auth.GetCredentials()
 
 	rt1UseXattrs := rt1.GetDatabase().UseXattrs()
-	_, err := rt2.RestTesterServerContext.AddDatabaseFromConfig(&DbConfig{
+	_, err = rt2.RestTesterServerContext.AddDatabaseFromConfig(&DbConfig{
 		BucketConfig: BucketConfig{
 			Server:   &server,
 			Bucket:   &bucketName,
@@ -954,24 +974,29 @@ func TestBulkDocsUnusedSequencesMultiRevDoc(t *testing.T) {
 	assertStatus(t, response, 201)
 
 	//Sequence 3 does not get used here as its using a different sequence allocator
-	doc21Rev, _ := rt2.GetDatabase().GetDocSyncData("bulk21")
+	doc21Rev, err := rt2.GetDatabase().GetDocSyncData("bulk21")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc21Rev.Sequence, uint64(4))
 
-	doc22Rev, _ := rt2.GetDatabase().GetDocSyncData("bulk22")
+	doc22Rev, err := rt2.GetDatabase().GetDocSyncData("bulk22")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc22Rev.Sequence, uint64(5))
 
-	doc23Rev, _ := rt2.GetDatabase().GetDocSyncData("bulk23")
+	doc23Rev, err := rt2.GetDatabase().GetDocSyncData("bulk23")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc23Rev.Sequence, uint64(6))
 
 	//Validate rev2 of doc "bulk1" has a new revision
-	doc1Rev2, _ := rt2.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev2, err := rt2.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev2.Sequence, uint64(7))
 
 	//Get the revID for doc "bulk1"
 	doc1RevID2 := doc1Rev2.CurrentRev
 
 	//Get current sequence number
-	lastSequence, _ = rt2.GetDatabase().LastSequence()
+	lastSequence, err = rt2.GetDatabase().LastSequence()
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, lastSequence, uint64(7))
 
 	//Now send a bulk_doc to rt1 to update doc bulk1 again
@@ -980,7 +1005,8 @@ func TestBulkDocsUnusedSequencesMultiRevDoc(t *testing.T) {
 	assertStatus(t, response, 201)
 
 	//Sequence 8 should get used here as sequence 3 should have been dropped by the first sequence allocator
-	doc1Rev3, _ := rt1.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev3, err := rt1.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev3.Sequence, uint64(8))
 
 	//validate the doc _sync metadata, should see last sequence lower than previous sequence
@@ -1003,17 +1029,20 @@ func TestBulkDocsUnusedSequencesMultiRevDoc2SG(t *testing.T) {
 	response := rt1.SendRequest("POST", "/db/_bulk_docs", input)
 	assertStatus(t, response, 201)
 
-	doc1Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev, err := rt1.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev.Sequence, uint64(1))
 
 	//Get the revID for doc "bulk1"
 	doc1RevID := doc1Rev.CurrentRev
 
-	doc3Rev, _ := rt1.GetDatabase().GetDocSyncData("bulk3")
+	doc3Rev, err := rt1.GetDatabase().GetDocSyncData("bulk3")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc3Rev.Sequence, uint64(2))
 
 	//Get current sequence number, this will be 3, as SG allocates enough sequences to process all bulk docs
-	lastSequence, _ := rt1.GetDatabase().LastSequence()
+	lastSequence, err := rt1.GetDatabase().LastSequence()
+	assertNoError(t, err, "LastSequence error")
 	assert.Equals(t, lastSequence, uint64(3))
 
 	rt2 := RestTester{RestTesterBucket: rt1.RestTesterBucket, SyncFn: `function(doc) {if(doc.type == "invalid") {throw("Rejecting invalid doc")}}`}
@@ -1030,7 +1059,7 @@ func TestBulkDocsUnusedSequencesMultiRevDoc2SG(t *testing.T) {
 	username, password, _ := spec.Auth.GetCredentials()
 
 	rt1UseXattrs := rt1.GetDatabase().UseXattrs()
-	_, err := rt2.RestTesterServerContext.AddDatabaseFromConfig(&DbConfig{
+	_, err = rt2.RestTesterServerContext.AddDatabaseFromConfig(&DbConfig{
 		BucketConfig: BucketConfig{
 			Server:   &server,
 			Bucket:   &bucketName,
@@ -1049,21 +1078,25 @@ func TestBulkDocsUnusedSequencesMultiRevDoc2SG(t *testing.T) {
 	assertStatus(t, response, 201)
 
 	//Sequence 3 does not get used here as its using a different sequence allocator
-	doc21Rev, _ := rt2.GetDatabase().GetDocSyncData("bulk21")
+	doc21Rev, err := rt2.GetDatabase().GetDocSyncData("bulk21")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc21Rev.Sequence, uint64(4))
 
-	doc22Rev, _ := rt2.GetDatabase().GetDocSyncData("bulk22")
+	doc22Rev, err := rt2.GetDatabase().GetDocSyncData("bulk22")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc22Rev.Sequence, uint64(5))
 
 	//Validate rev2 of doc "bulk1" has a new revision
-	doc1Rev2, _ := rt2.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev2, err := rt2.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev2.Sequence, uint64(7))
 
 	//Get the revID for doc "bulk1"
 	doc1RevID2 := doc1Rev2.CurrentRev
 
 	//Get current sequence number
-	lastSequence, _ = rt2.GetDatabase().LastSequence()
+	lastSequence, err = rt2.GetDatabase().LastSequence()
+	assertNoError(t, err, "LastSequence error")
 	assert.Equals(t, lastSequence, uint64(7))
 
 	//Now send a bulk_doc to rt1 to update doc bulk1 again
@@ -1072,7 +1105,8 @@ func TestBulkDocsUnusedSequencesMultiRevDoc2SG(t *testing.T) {
 	assertStatus(t, response, 201)
 
 	//Sequence 8 should get used here as sequence 3 should have been dropped by the first sequence allocator
-	doc1Rev3, _ := rt1.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev3, err := rt1.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev3.Sequence, uint64(8))
 
 	//Get the revID for doc "bulk1"
@@ -1084,7 +1118,8 @@ func TestBulkDocsUnusedSequencesMultiRevDoc2SG(t *testing.T) {
 	assertStatus(t, response, 201)
 
 	//Sequence 9 should get used here as sequence 6 should have been dropped by the second sequence allocator
-	doc1Rev4, _ := rt1.GetDatabase().GetDocSyncData("bulk1")
+	doc1Rev4, err := rt1.GetDatabase().GetDocSyncData("bulk1")
+	assertNoError(t, err, "GetDocSyncData error")
 	assert.Equals(t, doc1Rev4.Sequence, uint64(9))
 
 	//validate the doc _sync metadata, should see last sequence lower than previous sequence

